vulture_os/toolkit/api_parser/sentinel_one_identity_graph/sentinel_one_identity_graph.py

```diff
@@ -0,0 +1,263 @@
+        self.session.headers.update({"Authorization": f"Bearer {self.sentinel_one_identity_graph_token}"})
+        return
+
+    def execute_query(self, url: str, data: dict, timeout: int = 10):
+        retry = 3
+
+        while retry > 0:
+            try:
+                response = self.session.post(
+                    url=url,
+                    json=data,
+                    timeout=timeout)
+                resp_json = response.json()
+
+            except (HTTPError, ConnectionError) as e:
+                logger.exception(f"[{__parser__}][execute_query]: HTTPError raised -- {url} - {data} -- {response.content}", extra={'frontend': str(self.frontend)})
+                raise e
+            except JSONDecodeError as e:
+                logger.exception(f"[{__parser__}][execute_query]: Impossible to decode json response -- {response.content}", extra={'frontend': str(self.frontend)})
+                retry -= 1
+                continue
+            except (TimeoutError, ReadTimeout, ConnectTimeout) as e:
+                logger.exception(f"[{__parser__}][execute_query]: TimeoutError we will retry a request soon -- retry {retry}/3 -- {response.content}", extra={'frontend': str(self.frontend)})
+                retry -= 1
+                continue
+            break
+
+        return resp_json
+
+    def get_itdr_events_ids(self, since: int, to: int, limit: int = 50):
+        query = """
+            fragment PageInfo on PageInfo {endCursor hasNextPage}
+            query GetAlerts($first:Int, $after:String, $filters:[FilterInput!]) {
+                alerts(first:$first after:$after filters:$filters) {
+                    totalCount
+                    pageInfo {...PageInfo}
+                    edges {
+                        cursor
+                        node {
+                            id
+                        }
+                    }
+                }
+            }
+        """
+
+        variables  = {
+            'first': limit, # This is the limit of API returns for one query
+            'filters' : [
+                {
+                    'fieldId': 'detectedAt',
+                    'dateTimeRange': {
+                        'start': since,
+                        'startInclusive': True,
+                        'end': to,
+                        'endInclusive': False
+                    }
+                }
+                # {'fieldId': 'detectionProduct', 'stringIn': {'values': ['Identity']}} # Could be ignorded, this way of API will be used for EDR as well
+                # -> this tends to weird API bugs in combination of detectedAt
+            ]
+        }
+
+        edges = []
+        ret = []
+
+        response = self.execute_query(
+            self.url,
+            data={'query': query, 'variables': variables})
+
+        alerts = response.get('data', {}).get('alerts', {})
+        edges.extend(alerts.get('edges', []))
+
+        while alerts.get('pageInfo', {}).get('hasNextPage', False):
```

Comment:
Try and check `evt_stop` during this kind of while to avoid running too long when asked to stop